### ISSUE TITLE
 Warning: Trying to access array offset on value of type bool in /.../wp-content/plugins/elementor-pro/modules/dynamic-tags/tags/post-featured-image.php on line 36

### ISSUE THREAD

**Issue by jontng** [elementor/elementor#22785]
### Prerequisites

- [X] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [X] The issue still exists against the latest stable version of Elementor.

### Description

If there is no featured image, the function get_post_thumbnail_id returns false. This response is not handled correctly in this method.

public function get_value( array $options = [] ) {
	$thumbnail_id = get_post_thumbnail_id();
	if ( $thumbnail_id ) {
		$image_data = [
			'id' => $thumbnail_id,
			'url' => wp_get_attachment_image_src( $thumbnail_id, 'full' )[0],
		];
	} else {
		$image_data = $this->get_settings( 'fallback' );
	}
	return $image_data;
}

The URL source must be checked first.

### Steps to reproduce

1. Create some posts
2. Create a Grid Loop and use Featured Image or Dynamic tag to get Featured Image

### Isolating the problem

- [ ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [ ] This bug happens with a Blank WordPress theme active ([Hello theme](https://wordpress.org/themes/hello-elementor/)).
- [ ] I can reproduce this bug consistently following the steps above.

### System Info

					
== Server Environment ==
	Operating System: Darwin
	Software: Apache/2.4.43 (Unix)
	MySQL version: MySQL Community Server (GPL) v5.7.28
	PHP Version: 8.1.9
	PHP Memory Limit: 256M
	PHP Max Input Vars: 4000
	PHP Max Post Size: 1000M
	GD Installed: Yes
	ZIP Installed: Yes
	Write Permissions: All right
	Elementor Library: Connected

== WordPress Environment ==
	Version: 6.2.2
	Site URL: http://dev.gbcamp.nu
	Home URL: http://dev.gbcamp.nu
	WP Multisite: No
	Max Upload Size: 300 MB
	Memory limit: 40M
	Max Memory limit: 256M
	Permalink Structure: /%postname%/
	Language: sv-SE
	Timezone: Europe/Stockholm
	Admin Email: jon@bozzanova.se
	Debug Mode: Inactive

== Theme ==
	Name: Hello Elementor
	Version: 2.7.1
	Author: Elementor Team
	Child Theme: No

== User ==
	Role: administrator
	WP Profile lang: sv_SE
	User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36

== Active Plugins ==
	Advanced Custom Fields PRO
		Version: 6.1.6
		Author: WP Engine

	Classic Editor
		Version: 1.6.3
		Author: WordPress Contributors

	Elementor
		Version: 3.13.4
		Author: Elementor.com

	Elementor Pro
		Version: 3.13.2
		Author: Elementor.com

	Instant Images
		Version: 5.3.1
		Author: Darren Cooney

	Loco Translate
		Version: 2.6.4
		Author: Tim Whitlock

	Polylang Pro
		Version: 3.3.3
		Author: WP SYNTEX

	Post Type Switcher
		Version: 3.2.1
		Author: Triple J Software, Inc.

	Regenerate Thumbnails Advanced
		Version: 2.4.1
		Author: ShortPixel

	WordPress Importer
		Version: 0.8.1
		Author: wordpressdotorg

	WP All Import Pro
		Version: 4.8.0
		Author: Soflyy

== Elementanvändning ==
	
	footer : 1
		button : 4
		container : 11
		copyright : 1
		divider : 4
		heading : 7
		icon-list : 1
		producer : 1
		text-editor : 2
	archive : 1
		archive-posts : 1
		container : 1
	header : 1
		button : 2
		container : 4
		nav-menu : 1
		theme-site-title : 1
	widget : 0
		button : 1
	wp-page : 2
		button : 6
		container : 28
		heading : 15
		icon-list : 1
		image : 1
		image-box : 3
		loop-carousel : 1
		text-editor : 7
	single-post : 2
		accommodation-details : 1
		button : 5
		container : 11
		divider : 1
		gallery : 1
		heading : 4
		pricelist-single : 1
		summary : 2
		theme-post-content : 1
		theme-post-featured-image : 1
		theme-post-title : 2
	loop-item : 1
		button : 1
		container : 1
		summary : 1
		theme-post-featured-image : 1
		theme-post-title : 1

== Inställningar ==
	
	cpt_support: post, page
	allow_tracking: yes

== Funktioner ==
	Custom Fonts: 0
	Custom Icons: 0

== Integrationer ==
	

== Elementor-experiment ==
	Optimized DOM Output: Aktiv
	Förbättrad inläsning av filer: Aktiv
	Förbättrad CSS-laddning: Aktiv
	Inline Font Icons: Aktiv
	Tillgänglighetsförbättringar : Aktiv
	Ytterligare anpassade brytpunkter: Aktiv
	admin_menu_rearrangement: Inaktiv som standard
	Flexbox Container: Aktiv
	Uppgradera Swiper-bibliotek: Aktiv
	Grid Container: Inaktiv
	Default to New Theme Builder: Inaktiv
	Hello Theme Header & Footer: Aktiv som standard
	Editor Top Bar: Inaktiv
	Landningssidor: Inaktiv
	Nested Elements: Aktiv
	Uppskjuten inläsning av bakgrundsbilder: Aktiv
	Sidövergångar: Inaktiv
	Notes: Inaktiv
	Loop: Aktiv
	Formulärinlämningar: Inaktiv
	Scroll Snap: Inaktiv
	Meny: Inaktiv

== Logg ==
	
Log: showing 17 of 172022-10-21 09:48:09 [info] elementor-pro::elementor_pro_updater Started 
2022-10-21 09:48:09 [info] Elementor Pro/Upgrades - _on_each_version Start  
2022-10-21 09:48:09 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2022-10-21 09:48:09 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.7.5',
  'to' => '3.7.7',
)]
2022-10-21 09:48:09 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.7.5',
  'to' => '3.7.7',
)]
2023-06-13 17:06:37 [info] elementor::elementor_updater Started 
2023-06-13 17:06:37 [info] Elementor/Upgrades - _on_each_version Start  
2023-06-13 17:06:37 [info] Elementor/Upgrades - _on_each_version Finished 
2023-06-13 17:06:37 [info] Elementor/Upgrades - _v_3_8_0_fix_php8_image_custom_size Start  
2023-06-13 17:06:37 [info] Elementor/Upgrades - _v_3_8_0_fix_php8_image_custom_size Finished 
2023-06-13 17:06:37 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor',
  'from' => '3.7.8',
  'to' => '3.13.4',
)]
2023-06-13 17:06:37 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor',
  'from' => '3.7.8',
  'to' => '3.13.4',
)]
2023-06-13 17:07:41 [info] elementor-pro::elementor_pro_updater Started 
2023-06-13 17:07:41 [info] Elementor Pro/Upgrades - _on_each_version Start  
2023-06-13 17:07:41 [info] Elementor Pro/Upgrades - _on_each_version Finished 
2023-06-13 17:07:41 [info] Elementor data updater process has been completed. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.7.7',
  'to' => '3.13.2',
)]
2023-06-13 17:07:41 [info] Elementor data updater process has been queued. [array (
  'plugin' => 'Elementor Pro',
  'from' => '3.7.7',
  'to' => '3.13.2',
)]

PHP: showing 9 of 9PHP: 2022-10-21 09:52:14 [warning X 6][/Users/jontang/Local Sites/devgbcampnu/app/public/wp-content/plugins/elementor/core/common/modules/ajax/module.php::171] Undefined array key "data" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-06-13 17:44:30 [error X 1][/Users/jontang/Local Sites/devgbcampnu/app/public/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::304] Uncaught Error: Call to a member function update_meta() on null in /Users/jontang/Local Sites/devgbcampnu/app/public/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php:304
Stack trace:
#0 /Users/jontang/Local Sites/devgbcampnu/app/public/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php(177): ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->save_conditions(7, Array)
#1 [internal function]: ElementorPro\Modules\ThemeBuilder\Classes\Conditions_Manager->ajax_save_theme_template_conditions(Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#2 /Users/jontang/Local Sites/devgbcampnu/app/public/wp-content/plugins/elementor/core/common/modules/ajax/module.php(175): call_user_func(Array, Array, Object(Elementor\Core\Common\Modules\Ajax\Module))
#3 /Users/jontang/Local Sites/devgbcampnu/app/public/wp-includes/class-wp-hook.php(308): Elementor\Core\Common\Modules\Ajax\Module->handle_ajax_request('')
#4 /Users/jontang/Local Sites/devgbcampnu/app/public/wp-includes/class-wp-hook.php(332): WP_Hook->apply_filters('', Array)
#5 /Users/jontang/Local Sites/devgbcampnu/app/public/wp-includes/plugin.php(517): WP_Hook->do_action(Array)
#6 /Users/jontang/Local Sites/devgbcampnu/app/public/wp-admin/admin-ajax.php(188): do_action('wp_ajax_element...')
#7 {main}
  thrown [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-06-13 21:00:49 [warning X 3][/Users/jontang/Local Sites/devgbcampnu/app/public/wp-content/plugins/elementor-pro/modules/theme-builder/classes/conditions-manager.php::171] Undefined array key "editor_post_id" [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-06-14 10:34:38 [warning X 3][/Users/jontang/Local Sites/devgbcampnu/app/public/wp-content/plugins/elementor/app/modules/import-export/module.php::717] Attempt to read property "label" on null [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-06-14 10:34:39 [warning X 3][/Users/jontang/Local Sites/devgbcampnu/app/public/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php::150] Undefined array key "condition_type" [array (
  'trace' => '
#0: /Users/jontang/Local Sites/devgbcampnu/app/public/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(150): Elementor\Core\Logger\Manager -> rest_error_handler()
#1: ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_template_json_item()
#2: /Users/jontang/Local Sites/devgbcampnu/app/public/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(120): class type array_map()
#3: /Users/jontang/Local Sites/devgbcampnu/app/public/wp-content/plugins/elementor-pro/core/app/modules/site-editor/data/endpoints/templates.php(59): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> normalize_templates_json()
#4: /Users/jontang/Local Sites/devgbcampnu/app/public/wp-content/plugins/elementor/data/base/endpoint.php(158): ElementorPro\Core\App\Modules\SiteEditor\Data\Endpoints\Templates -> get_items()
',
)]
PHP: 2023-06-14 10:43:30 [warning X 6][/Users/jontang/Local Sites/devgbcampnu/app/public/wp-content/plugins/elementor-pro/modules/dynamic-tags/acf/tags/acf-url.php::34] Undefined array key 1 [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-06-15 11:06:40 [warning X 50][/Users/jontang/Local Sites/devgbcampnu/app/public/wp-content/plugins/elementor-pro/modules/dynamic-tags/tags/post-featured-image.php::36] Trying to access array offset on value of type bool [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-06-15 11:07:37 [warning X 1][/Users/jontang/Local Sites/devgbcampnu/app/public/wp-content/plugins/elementor/modules/history/revisions-manager.php::363] Trying to access array offset on value of type null [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]
PHP: 2023-06-15 11:57:31 [warning X 1][/Users/jontang/Local Sites/devgbcampnu/app/public/wp-content/plugins/elementor-pro/modules/posts/skins/skin-content-base.php::265] Attempt to read property "ID" on null [array (
  'trace' => '
#0: Elementor\Core\Logger\Manager -> shutdown()
',
)]

JS: showing 2 of 2JS: 2023-06-13 19:09:41 [error X 2][http://dev.gbcamp.nu/wp-content/plugins/elementor/assets/js/frontend-modules.min.js?ver=3.13.4:2:3583] Cannot read properties of undefined (reading 'params') 
JS: 2023-06-14 08:45:03 [error X 1][http://dev.gbcamp.nu/wp-content/plugins/elementor/assets/js/common-modules.min.js?ver=3.13.4:2:55595] Model.isValidChild() should be implemented, please provide 'isValidChild' functionality. 

== Elementor - Compatibility Tag ==
	
	Elementor Pro: Kompatibel

== Elementor Pro - Compatibility Tag ==

**Comment by nicholaszein** [comment]
Hello @jontng! 

Thank you for submitting your issue! 🙏 

Please allow us some time to investigate it, and get back to you with our findings. 🧐 

Best regards

**Comment by digitalmurph** [comment]
I'm having exactly the same issue. Any solution to this yet? Thanks

**Comment by Erudition** [comment]
Same issue but with other missing fields, such as Pods fields.
```
Undefined array key "data" in /var/www/html/web/wp/wp-content/plugins/elementor/core/common/modules/ajax/module.php on line 175

 Trying to access array offset on value of type bool in /var/www/html/web/wp/wp-content/plugins/elementor-pro/modules/dynamic-tags/pods/tags/pods-text.php on line 20
```

Possible workaround:
go to source code of Elementor Pro: elementor-pro/modules/dynamic-tags/pods/tags/pods-text.php => Edit (on line 20): $field = $field_data[‘field’]; to $field = $field_data[‘field’] ?? 0;

Or equivalent for featured image module.

**Comment by digitalmurph** [comment]
Any progress on this problem yet?
Thanks

**Comment by dulakshirb** [comment]
Have you found any solution for this by chance?

**Comment by mrmasstransit** [comment]
I can also report that I am having this issue on Elementor 3.16.4 / Elementor Pro 3.16.2

**Comment by jamieburchell** [comment]
2024 here, still not fixed

```
Trying to access array offset on value of type bool in /var/www/web/app/plugins/elementor-pro/modules/dynamic-tags/tags/post-featured-image.php on line 39
```

@nicholaszein

**Comment by oscarmat** [comment]
same error:
Trying to access array offset on false in (....) /wp-content/plugins/elementor-pro/modules/dynamic-tags/tags/post-featured-image.php on line 39.

Elementor 3.21.4 
Elementor Pro 3.2.1.2

**Comment by FrankWalle** [comment]
Hi.
Any progress on this?
Trying to access array offset on value of type bool in /var/www/vhosts/xxxx/httpdocs/wp-content/plugins/elementor-pro/modules/dynamic-tags/tags/post-featured-image.php on line 39'

Any idea? This really kills the cpu.

**Comment by floodwayprintco** [comment]
Hi there, has there been any movement on this? I am experiencing the same issue, I think it is causing a major break in my page.

I am not opposed to adjusting my products. If there is an ID that needs to be set before import or something. Any suggestions?

**Comment by stagelync-laci** [comment]
Hi, I have the same issue. When can we expect it to be fixed?

**Comment by geekvibesnation** [comment]
I am also having issues with this using FIFU and getting /wp-content/plugins/elementor-pro/modules/dynamic-tags/tags/post-featured-image.php on line 39

Hopefully a fix can be found soon
